### DISCUSSION TITLE
Optional mount parameters are not resolved with default

### DISCUSSION THREAD

**Discussion by svenbw** [livewire/livewire#5723] [7. Bugs (archived)]
I have created a Livewire component with two parameters, these parameters are resolved to uuid's which are public 
(this allows me to pass models and strings (uuid) to the component, which keeps the upper html code "clean").
```php
public function mount(FirstModel|string|null $one = null, SecondModel|string $two)
{
    if ($one === null) {
        $this->create();
    } elseif (is_string($one)) {
        $this->oneUuid = $one;
    } else {
        $this->oneUuid = $one->uuid;
    }
}
```

In the upper blade template I create the model leaving out the first mount parameter
```html
<livewire:my-component
    :two="$modelTwo"
/>
```

When creating the Livewire component the creation fails: _Unable to resolve dependency [Parameter #0 [ <required> App\Models\FirstModel|string|null $one ]]_ which makes sense... This parameter is not there so the error makes sense, however there IS a default that could be picked up.

Changing the code to this works fine: Here one will be resolved as a (non existing) model:
```php
public function mount(FirstModel $one, SecondModel|string $two)
{
    if ( ! $one->existsl) {
        $this->create();
    } else {
        $this->oneUuid = $one->uuid;
    }
}
```

If the argument becomes nullable the (non existing) model is still passed:
```php
public function mount(?FirstModel $one = null, SecondModel|string $two)
{
   \\ -> $one will still be of the FirstModel type if omitted during creation.
```

I'm still looking into the details but the first issue seems to be caused by the fact that `Livewire\ImplicitlyBoundMethod\getParameterClassName` checks if the parameters are an instance of `ReflectionNamedType`, but they are an instance of `ReflectionUnionType`.

However, the might not be an issue, depending on the answers of the next questions:
- Is it allowed to specify multiple classes in a mount?
- ~~Should a nullable parameter assign null to the parameter?~~

**Context**
    Livewire version: v2.12.3
    Laravel version: v10.13.0
    PHP: v8.2.6

**Comment by svenbw** [comment]
First thing to note: `public function mount(FirstModel|string|null $one = null, SecondModel|string $two)` is wrong, optional parameters should be added to the end `public function mount(SecondModel|string $two, FirstModel|string|null $one = null)`
This makes the second question "Should a nullable parameter assign null to the parameter?" not applicable.

**Comment by eliezerben** [comment]
I'm also looking for a solution to this. How can I have optional parameters in mount?
